crates/trigger/src/cli/initial_kv_setter.rs

```diff
@@ -23,9 +23,12 @@ impl<F: RuntimeFactors, U> ExecutorHooks<F, U> for InitialKvSetterHook {
         &mut self,
         configured_app: &spin_factors::ConfiguredApp<F>,
     ) -> anyhow::Result<()> {
-        let Some(kv) = configured_app.app_state::<KeyValueFactor>().ok() else {
+        if self.kv_pairs.is_empty() {
             return Ok(());
-        };
+        }
+        let kv = configured_app
+            .app_state::<KeyValueFactor>()
+            .context("attempted to set initial kv pairs but there is no kv")?;
```

Comment:
This would only ever happen if the runtime is misconfigured so I think it's better to provide a more technical error message since only runtime embedders should see this (and end users should never encounter this).

```suggestion
            .context("attempted to set initial kv pairs but the key-value factor was not configured")?;
```